### DISCUSSION TITLE
tree-sitter generate: No such file or directory (os error 2)

### DISCUSSION THREAD

**Discussion by matoous** [tree-sitter/tree-sitter#1588] [Q&A] [answered]
Hello! I tried to follow the guide at https://tree-sitter.github.io/tree-sitter/creating-parsers to create new syntax but am running into issues. It might be something trivial that I missing, apologies in advance.

I created the project including basic structure but when I try to run `tree-sitter generate` it fails with `No such file or directory (os error 2)`. I can circumvent this by running `tree-sitter generate grammer.js` instead.

But even after that, when I try `tree-sitter parse example-file` it fails with:
```
> tree-sitter parse example-file

Warning: You have not configured any parser directories!
Please run `tree-sitter init-config` and edit the resulting
configuration file to indicate where we should look for
language grammars.

Failed to load language in current directory

Caused by:
    0: Failed to read grammar.json
    1: No such file or directory (os error 2)
```

Running `tree-sitter init-config` creates file at my home directory (`/Users/name/Library/Application Support/tree-sitter/config.json`) but even after that I have no luck:
```
> tree-sitter parse example-file

Failed to load language in current directory

Caused by:
    0: Failed to read grammar.json
    1: No such file or directory (os error 2)
```

The code is here: https://github.com/matoous/tree-sitter-jsonnet, I double checked that everything is setup accordingly to other working syntaxes but no luck so far.

**Comment by maxbrunsfeld** [comment, marked as answer]
I think it's just because your file is called `grammer.js` instead of `grammar.js`.

**Reply by matoous** [reply]
Oh man 🙈 now I feel stupid. Thanks a lot for the catch!
